datalab/datalab_session/data_operations/median.py

```diff
@@ -31,4 +37,50 @@ def wizard_description():
         }
     
     def operate(self):
-        pass
+        input_files = self.input_data.get('input_files', [])
+        completion_total = len(input_files)
+        image_data_list = []
+
+        log.info(f'Executing median operation on {completion_total} files')
+
+        # fetch fits for all input data
+        for index, file_info in enumerate(input_files):
+            basename = file_info.get('basename', 'No basename found')
+            
+            fits_file = find_fits(basename)
+            fits_url = fits_file[0].get('url', 'No URL found')
+
+            with fits.open(fits_url, use_fsspec=True) as hdu_list:
+                data = hdu_list['SCI'].data
+                image_data_list.append(data)
+                self.set_percent_completion((index) / completion_total)
+
+        # Crop fits image data to be the same shape then stack 
+        min_shape = min(arr.shape for arr in image_data_list)
+        cropped_data_list = [arr[:min_shape[0], :min_shape[1]] for arr in image_data_list]
+        stacked_data = np.stack(cropped_data_list, axis=2)
+
+        # Calculate a Median along the z axis
+        median = np.median(stacked_data, axis=2)
+
+        # Create a new Fits File
+        cache_key = self.generate_cache_key()
+        hdr = fits.Header([('KEY', cache_key)])
+        primary_hdu = fits.PrimaryHDU(header=hdr)
```

Comment:
You will probably want some more things in the header like NAXIS1/2 and stuff like that, but Matt would probably know better.

Reply:
@mgdaily

Reply:
I think it's ok to have a minimal header at the moment. Let's add things as we need them